src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/mappers/DateMapper.java

```diff
@@ -46,7 +46,10 @@ public static int secondsSinceStartOfService(
     ZonedDateTime dateTime,
     ZoneId zoneId
   ) {
-    ZonedDateTime startOfService = asStartOfService(departureDate.toLocalDate(), zoneId);
+    ZonedDateTime startOfService = asStartOfService(
```

Comment:
I think it should be pretty easy to write some tests for this method.

Reply:
I added new unit test for the edge case.